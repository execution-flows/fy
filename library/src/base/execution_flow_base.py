# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
#  file, You can obtain one at https://mozilla.org/MPL/2.0/.

import abc
from typing import Generic, TypeVar

ResultT = TypeVar("ResultT")


class ExecutionFlowBase(Generic[ResultT], abc.ABC):
    @abc.abstractmethod
    def __call__(self) -> ResultT:
        raise NotImplementedError()
