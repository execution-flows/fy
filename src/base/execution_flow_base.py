import abc
from typing import Generic, TypeVar

ResultT = TypeVar("ResultT")


class ExecutionFlowBase(Generic[ResultT], abc.ABC):
    @abc.abstractmethod
    def __call__(self) -> ResultT:
        raise NotImplementedError()
