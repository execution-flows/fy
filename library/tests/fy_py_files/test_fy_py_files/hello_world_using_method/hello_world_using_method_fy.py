"""fy
flow HelloWorld -> None:
    method greet using constant
"""

from base.flow_base import FlowBase

from fy_py_files.test_fy_py_files.hello_world_using_method.using_constant_fy import (
    Greet_UsingConstant_MethodMixin,
)


# fy:start <<<===
class HelloWorld_Flow(
    # Method Mixins
    Greet_UsingConstant_MethodMixin,
    # Base
    FlowBase[None]
):
    def __call__(self) -> None:
        # fy:end <<<===
        self._greet()
