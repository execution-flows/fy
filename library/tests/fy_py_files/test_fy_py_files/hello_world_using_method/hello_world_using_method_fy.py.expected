"""fy
base flow HelloWorld -> None:
    method greet using constant
"""

import abc
from fy_core.base.flow_base import FlowBase
from fy_py_files.test_fy_py_files.hello_world_using_method.using_constant_fy import (
    Greet_UsingConstant_MethodMixin,
)


# fy:start ===>>>
class HelloWorld_BaseFlow(
    # Method Mixins
    Greet_UsingConstant_MethodMixin,
    # Base
    FlowBase[None],
    abc.ABC,
):
    pass
    # fy:end <<<===
